### PR TITLE
[GPU] Check memory allocation size of network graph

### PR DESCRIPTION
**Pull request by jade-cho**
### Check memory allocation sizes of device and host:
- Throw exception if network total alloc > max (total_device_mem, total_host_mem)
### Tickets:
- 74243

### CODE REVIEW

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp [hunk 1]
```diff
@@ -94,6 +94,15 @@ memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type ty
         IE_THROW() << ss.str();
     }
 
+    auto used_mem = get_used_device_memory(allocation_type::usm_device) + get_used_device_memory(allocation_type::usm_host);
+    if (layout.bytes_count() + used_mem > get_max_memory_size()) {
+        std::stringstream ss;
+        ss << "Exceeded max size of memory allocation: "
+            << "Required " << layout.bytes_count() + used_mem << " bytes "
+            << "but max alloc size is " << get_max_memory_size() << " bytes";
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp:102]: `but max alloc size is` -> `but memory size is`

**Comment by isanghao** [comment]
LGTM. Please see minor comment.

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp (hunk 1 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp:102]: It will be changed.

**Comment by yeonbok** [comment]
What kind of issue is this PR targeting? 
Mostly, the host mem is much bigger than device mem, so it will check with host mem size. 
If it is targeting OoO memory killer (linux), it would be enough to apply it for linux only? 
If I remember correctly, windows does not do ooo memory killer.  (correct me if I am wrong).
Also it would be great to describe  the target situation. For me it was ambiguous that checking any types of allocation with max of host and device mem.

**Comment by jade-cho** [comment]
This PR is for that linux system is hanged even though the OOM killer is called. It will not be merged for now. I need to discuss the max mem size is calculated correctly later.

**Review by isanghao — APPROVED** [review]

**Pull request merged**